### PR TITLE
Updates people and places APIs to minify contacts

### PR DESCRIPTION
**Pull request by garethbowen**
The people and places APIs were missed from the Linked Contacts refactor so were still creating inlined parents and contacts.

medic/medic-webapp#2635

### CODE REVIEW

**Comment by garethbowen** [comment]
@mandric Review if you please? There's a heap of refactoring so you're better off reviewing each commit separately.

**Comment by mandric** [comment]
It's worth noting here how we use the word "refactor".  Refactor to me means major code logic changes or redesigning code and has nothing to do with reformatting or style (ES6 shorthand).  Also when you use "refactor" in the title of a commit I'd expect a body to explain in plain english what the refactor is, highlighting the rationale behind the refactor.  So I was immediately confused by two things in the first commit, straight away, simply because of a terminology issue.  It might be worth documenting our definitions for "reformat" and "refactor".  I hate to get bogged down in minor process adjustments but you put in effort to split up the commits nicely, maybe we should get the wording down and crystalize it!

File: controllers/places.js [hunk 1]
```diff
@@ -252,3 +269,4 @@ module.exports.createPlace = createPlaces;
 module.exports.getPlace = getPlace;
 module.exports.updatePlace = updatePlace;
 module.exports.getOrCreatePlace = getOrCreatePlace;
+module.exports.minify = minify;
```
Comment (mandric) [inline controllers/places.js]: Are we still doing the export using `_` (underscore prefix) if it's a private function thing (only exported for testing).

File: controllers/people.js [hunk 2]
```diff
@@ -24,12 +24,10 @@ var getPerson = function(id, callback) {
   });
 };
 
-var isAPerson  = function(obj) {
-  return obj.type === 'person';
-};
+const isAPerson = obj => obj.type === 'person';
```
Comment (garethbowen) [inline controllers/people.js]: One liners auto-return.

File: controllers/places.js (hunk 1 above)
Comment (garethbowen) [inline controllers/places.js]: `minify` is used outside this module so it shouldn't have the underscore. Soon we can extract it into your library but for now this is a decent place to export it from.

IMO we should stop using the underscore prefix convention. My preferred way is to break each module up so it's small enough to unit test without having to expose the guts of it. If that's not practical then the next best way is to export more function when the TEST_ENV is set. That way you can't accidentally use one in production code.

**Comment by garethbowen** [comment]
To me, refactoring is simply any change that doesn't change the external behaviour.
